lib/active_model/serializer/reflection.rb

```diff
@@ -78,6 +78,17 @@ def build_association(subject, parent_serializer_options)
 
       private
 
+      def polymorphic_key(assoc_value)
+        reflection_key = assoc_value.class.model_name
+        assoc_value.respond_to?(:each) ? reflection_key.plural : reflection_key.singular
+      end
+
+      def reflection_options(assoc_value, options)
+        reflection_options = options.dup
+        reflection_options[:key] = polymorphic_key(assoc_value) if options[:polymorphic] && !options.key?(:key)
```

Comment:
Note: We might want to make it a `||=` in order not to override user-provided options.